### PR TITLE
Mini challenge 3 (Implement searching videos feature)

### PR DESCRIPTION
**Pull request by adrianchavez123**
This pr covers:
- Include searching videos feature.
- Created new components to visualize a video
- Created new hooks to query youtube data api
- add basic test cases
- incorporate context to reduce the amount of  youtube data api calls

The following image displays the code coverage:
image

### CODE REVIEW

**Comment by adrianchavez123** [comment]
The testing on searchingBar component is quite low and lowering the average coverage.
I started using context but it is somewhat related to next mini-challenge.

File: src/components/Header/SearchingBar/SearchingBar.component.jsx [hunk 1]
```diff
@@ -1,10 +1,66 @@
...
+import { faSearch } from '@fortawesome/free-solid-svg-icons';
+import { Search, SearchContainer, Label } from './SearchingBar.styles';
+import { useVideo } from '../../../providers/Video';
 
 function SearchingBar() {
+  const { search, setSearch, setVideos, setAlert } = useVideo();
+  const history = useHistory();
+  const { videoId } = useParams();
+
+  const searchVideos = async (e) => {
```
Comment (PacoMojica) [inline src/components/Header/SearchingBar/SearchingBar.component.jsx:13]: This code ca be improved, I think the function is doing quite a bit, maybe create a custom hook to search can be a great idea

File: src/components/VideoPlayer/VideoPlayer.component.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  VideoRating,
+} from './VideoPlayer.styles';
+import VideoWrapper from '../VideoWrapper/VideoWrapper.component';
+
+function VideoPlayer({ videoId }) {
+  const { setAlert } = useVideo();
+  const { video, isLoading, error } = useVideoPlayerApi({
+    type: 'VIDEO_DETAILS',
+    payload: videoId,
+  });
```
Comment (PacoMojica) [inline src/components/VideoPlayer/VideoPlayer.component.jsx:25]: This looks great, you mentioned that you are working on the search to use context, which is something for required for the next challenge, you could use something like this hook for the search request

File: src/pages/Home/Home.page.jsx [hunk 3]
```diff
@@ -1,13 +1,28 @@
...
+  const { videoId } = useParams();
+  let content = (
+    <>
       <h1>Welcome to the Challenge!</h1>
       <VideoList />
+    </>
+  );
+  if (videoId) {
+    content = <VideoPlayer videoId={videoId} />;
+  }
```
Comment (PacoMojica) [inline src/pages/Home/Home.page.jsx:20]: you can create a page dedicated to the video player

**Review by PacoMojica — COMMENTED** [review]
Hi Adrian, you challenge looks great, remember testing the events on the components that require it, by doing so you can improve the branch percentage

## Acceptance Criteria
- [x] Videos in the Home View are fetched from the YouTube API and displayed correctly.
- [x] Users can search for YouTube videos using the search field on the header.
- [x] A video can be played from the **Video Details View** after clicking on it.
- [x] The video information and related videos list are displayed correctly on the **Video Details View**.
- [x] When a user clicks on a related video the video player, information and related videos list are updated.

### Bonus Points
- [x] Custom Hooks for API calls are implemented and tested correctly.
- [x] Testing coverage is above 60%. (Please include a screenshot of the code coverage report in your PR description).

File: src/pages/Home/Home.page.jsx (hunk 3 above)
Comment (adrianchavez123) [reply src/pages/Home/Home.page.jsx:20]: Cool, I'm going to try that. the instructions' page suggested conditional rendering but i can move the component to a separe page.

File: src/components/VideoPlayer/VideoPlayer.component.jsx (hunk 2 above)
Comment (adrianchavez123) [reply src/components/VideoPlayer/VideoPlayer.component.jsx:25]: of course, that was my idea use it on `useVideoPlayerApi ` for the search and video details page. but the response (returned values) were slightly different for one i need a single video and for search a list

File: src/components/Header/SearchingBar/SearchingBar.component.jsx (hunk 1 above)
Comment (adrianchavez123) [reply src/components/Header/SearchingBar/SearchingBar.component.jsx:13]: agree

**Comment by adrianchavez123** [comment]
I'm going to close this and include the changes on the next pr ( https://github.com/adrianchavez123/react-certification-2020/pull/4)

**Pull request closed without merging**